internal/config/global.go

```diff
@@ -1,91 +1,141 @@
+func (u *UrlValue) Type() string {
+	return "url"
 }
 
-func (g *GlobalOptions) AddFlags(fs *flag.FlagSet) {
-	fs.Func("rekor-url", "Rekor server URL", func(s string) error {
-		u, err := url.ParseRequestURI(s)
-		if err != nil {
-			return err
-		}
-		g.RekorUrl = u.String()
-		return nil
-	})
-
-	fs.Func("fulcio-url", "Fulcio server url", func(s string) error {
-		u, err := url.ParseRequestURI(s)
-		if err != nil {
-			return err
-		}
-		g.FulcioUrl = u.String()
-		return nil
-	})
-
-	fs.Func("oidc-issuer-url", "OIDC issuer url for keyless signing", func(s string) error {
-		u, err := url.ParseRequestURI(s)
-		if err != nil {
-			return err
-		}
-		g.OidcIssuerUrl = u.String()
-		return nil
-	})
-
-	fs.StringVar(&g.OidcClientId, "oidc-client-id", "sigstore", "OIDC client id for keyless signing")
-	fs.StringVar(&g.ArtifactDigest.Value, "artifact-digest", "", "Digest of the OCI artifact. Should be prefixed with the digest hash type, e.g., sha256:60bcfdd2...")
-	fs.StringVar(&g.ArtifactUri, "artifact-uri", "", "URI of the OCI artifact")
-	fs.StringVar(&g.IdToken, "id-token", "", "ID token to use for keyless signing")
+func (u *UrlValue) Value() *url.URL {
+	return u.value
 }
 
-func (g *GlobalOptions) KeyOpts() options.KeyOpts {
-	return options.KeyOpts{
-		IDToken:      g.IdToken,
-		OIDCIssuer:   g.OidcIssuerUrl,
-		OIDCClientID: g.OidcClientId,
-		FulcioURL:    g.FulcioUrl,
-		RekorURL:     g.RekorUrl,
+type DigestValue struct {
+	Hex       string
+	Algorithm string
+}
+
+func (d *DigestValue) Set(digest string) error {
+	var found bool
+	d.Algorithm, d.Hex, found = strings.Cut(digest, ":")
+	if !found {
+		return fmt.Errorf("expected artifact digest to be of the form hashType:digestValue")
+	}
+
+	return nil
+}
+
+func (d *DigestValue) String() string {
+	if d.Algorithm == "" && d.Hex == "" {
+		return ""
 	}
+
+	return d.Algorithm + ":" + d.Hex
+}
+
+func (d *DigestValue) Type() string {
+	return "digest"
 }
 
-func GetGitHubEnvToken() (string, error) {
-	githubToken, githubTokenExists := os.LookupEnv("GITHUB_TOKEN")
+func NewGlobalOptions() GlobalOptions {
+	fulcioUrl, _ := url.ParseRequestURI("https://fulcio.sigstore.dev")
+	rekorUrl, _ := url.ParseRequestURI("https://rekor.sigstore.dev")
+	issuerUrl, _ := url.ParseRequestURI("https://oauth2.sigstore.dev/auth")
 
-	if !githubTokenExists {
-		return "", fmt.Errorf("GITHUB_TOKEN not provided")
+	return GlobalOptions{
+		FulcioUrl:      &UrlValue{value: fulcioUrl},
+		RekorUrl:       &UrlValue{value: rekorUrl},
+		OidcIssuerUrl:  &UrlValue{value: issuerUrl},
+		ArtifactDigest: &DigestValue{},
 	}
+}
 
-	return githubToken, nil
+func (g *GlobalOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&g.ArtifactUri, "artifact-uri", "", "URI of the OCI artifact")
+	cobra.CheckErr(cmd.MarkFlagRequired("artifact-uri"))
+
+	cmd.Flags().Var(g.ArtifactDigest, "artifact-digest", "Digest of the OCI artifact. Should be prefixed with the digest hash type, e.g., sha256:abc123")
```

Comment:
Cobra supports flags that are inherited from the parent command (persistent flags), which I considered using for the global options, but I think it's a bit cleaner and more flexible with the embedded struct.